compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -398,3 +468,60 @@ pub fn create_compressed_metadata_file(
 
     file.write().unwrap()
 }
+
+/// * Xcoff - On AIX, custom sections are merged into predefined sections,
+///   so custom .rustc section is not preserved during linking.
+///   For this reason, we store metadata in predefined .info section, and
+///   define a symbol to reference the metadata. To preserve metadata during
+///   linking on AIX, we have to
+///   1. Create an empty .text section, a empty .data section.
+///   2. Define an empty symbol named `symbol_name` inside .data section.
+///   3. Define an symbol named `AIX_METADATA_SYMBOL_NAME` referencing
+///      data inside .info section.
+///   From XCOFF's view, (2) creates a csect entry in the symbol table, the
+///   symbol created by (3) is a info symbol for the preceding csect. Thus
+///   two symbols are preserved during linking and we can use the second symbol
+///   to reference the metadata.
+pub fn create_compressed_metadata_file_for_xcoff(
+    mut file: write::Object<'_>,
+    data: &[u8],
+    symbol_name: &str,
+) -> Vec<u8> {
+    assert!(file.format() == BinaryFormat::Xcoff);
+    file.add_section(Vec::new(), b".text".to_vec(), SectionKind::Text);
```

Comment:
Is the `.text` section mandatory? It doesn't seem to be used.

Reply:
This is to satisfy AIX system linker's behavior, it aborts by fatal error if it meets a valid XCOFF file with no `.text`, no `.data` and no `.bss`.

Reply:
Could you add a comment to that effect in both functions?